### PR TITLE
PR 3 of n: Add inline comments where users need to replace code

### PR DESCRIPTION
**Pull request by cbethell**
## Purpose:
What issue(s) does your PR address?

This PR addresses issue #99  by adding inline comments in the notebooks in this repository where users would need to replace code to tailor the notebooks for use with different data files.

This PR adds inline comments to the notebooks in the pathway-analysis directory.

Please let me know if I missed any sections that should have an inline comment added or if the wording I added is not clear.

Note that I defined the `data_dir` object here in this PR (`data_dir <- "data"`), but I did not add a comment telling the user to replace its file path as I believe we prefer to urge users to store their data in the `data` directory (if I am not mistaken). This is something that I handled differently in PR #104 and #105 (where I didn't declare the data directory file path as an object and add a comment telling users to replace the path with the path to where their data file is stored). I handled it different in this PR so we can get an idea of the which we would prefer. 
(cc: @cansavvy as this is related to her [comment](https://github.com/AlexsLemonade/refinebio-examples/pull/104#discussion_r451038542) on the first PR)

Also note that the `results/Project_GSE37418_up_SHH_lfc2_fdr0_05` folder automatically saves with the `_` instead of a `.` for the `0.05` at the end of that file name. Therefore, I removed the old folder that was named `results/Project_GSE37418_up_SHH_lfc2_fdr0.05` and made the appropriate changes to the `pathway-analysis/ora_with_webgestaltr.Rmd ` notebook. (I ran this a few times to ensure that it was not something wrong with my environment but I keep getting the project folder named `results/Project_GSE37418_up_SHH_lfc2_fdr0_05` when I rerun the notebook)

## Results
If applicable, briefly report your findings.

This PR includes updates to the results files in the pathway-analysis directory, which again I believe reflect updates made to the input files (so this should be fine but correct me if I am wrong).

## Pull Request Check List:
* [x] Run a linter
* [x] Set the seed (if applicable)
* [x] Comments and/or documentation up to date
* [x] Double check your paths
* [x] Check headline numbering
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

### CODE REVIEW

File: pathway-analysis/kegg_ortholog_mapping.Rmd [hunk 1]
```diff
@@ -184,7 +184,8 @@ Let's pick the polymerase pathway, where our success should be pretty obvious
 from the mouse gene symbols.
 
 ```{r}
-kegg_mouse_list[[grep("POLYMERASE", names(kegg_mouse_list))]]
+kegg_mouse_list[[grep("POLYMERASE", # Replace with desired pathway
```
Comment (cansavvy) [inline pathway-analysis/kegg_ortholog_mapping.Rmd:187]:
```suggestion
kegg_mouse_list[[grep("POLYMERASE", # Replace with an exact word or phrase that would filter in your desired pathway(s)
```

File: pathway-analysis/ora_with_webgestaltr.Rmd [hunk 2]
```diff
@@ -80,7 +84,7 @@ We can identify upregulated genes by filtering these results and use this list
 as input to ORA.
 
 ```{r}
-dge_df <- readr::read_tsv("https://github.com/AlexsLemonade/refinebio-examples/raw/10b116dff0d44267db3e3a6fde735187e1bcd195/validate-differential-expression/results/GSE37418_limma_results.tsv")
+dge_df <- readr::read_tsv("https://github.com/AlexsLemonade/refinebio-examples/raw/10b116dff0d44267db3e3a6fde735187e1bcd195/validate-differential-expression/results/GSE37418_limma_results.tsv") # Replace with path or link to your desired differential expression results TSV file
```
Comment (cansavvy) [inline pathway-analysis/ora_with_webgestaltr.Rmd:87]: To make this clearer to the user, maybe declare this on the line before and call it something like `de_url` and then make it clear that it has to be a link to a TSV file OR a regular file path to a TSV one would have on their computer.

Comment (cansavvy) [reply pathway-analysis/ora_with_webgestaltr.Rmd:87]: Second question is do we want this to be a permalink to this particular file? or do we want to make it so its whatever the file is in the current release? Permalink is probably fine for now, but we could also revisit this later.

File: pathway-analysis/ora_with_webgestaltr.Rmd [hunk 3]
```diff
@@ -148,14 +152,15 @@ go_enrichment_results <-
...
+              projectName = "GSE37418_up_SHH_lfc2_fdr0_05" # Replace with a relevant output name for the HTML file
+              )
 ```
 
 As noted in the messages from `WebGestaltR`, the results are in
-`results/Project_GSE37418_up_SHH_lfc2_fdr0.05`.
+`results/Project_GSE37418_up_SHH_lfc2_fdr0_05`.
 The HTML report is at 
-[`results/Project_GSE37418_up_SHH_lfc2_fdr0.05/Report_GSE37418_up_SHH_lfc2_fdr0.05.html`](./results/Project_GSE37418_up_SHH_lfc2_fdr0.05/Report_GSE37418_up_SHH_lfc2_fdr0.05.html).
+[`results/Project_GSE37418_up_SHH_lfc2_fdr0.05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html`](./results/Project_GSE37418_up_SHH_lfc2_fdr0.05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html).
```
Comment (cansavvy) [inline pathway-analysis/ora_with_webgestaltr.Rmd:163]:
```suggestion
[`results/Project_GSE37418_up_SHH_lfc2_fdr0.05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html`](./results/Project_GSE37418_up_SHH_lfc2_fdr0.05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html).
```

This file name looks odd to me (the part with the `fdr0_05`. But if that's what the file name is, I guess it's fine?

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 4]
```diff
@@ -188,16 +199,23 @@ We need to unzip it here if we have not already.
 
 ```{r}
 # if the unzipped folder does not exist
-if (!dir.exists(file.path("data", "GSE37382"))) {
-  compressed_file <- file.path("data", "GSE37382.zip")
-  unzip(compressed_file, exdir = "data")
+if (!dir.exists(file.path(data_dir,
+                          "GSE37382" # Replace with the name of your file without the .zip extension
+                          ))) {
+  compressed_file <- file.path(data_dir,
```
Comment (cansavvy) [inline pathway-analysis/qusage_meta_analysis.Rmd:205]: We should probably provide a bit more context around this zip file. I think its unlikely the user will have a zip file, but we may want to explain they should skip/delete this step if they don't have a zip file (which I don't think they will).

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 5]
```diff
@@ -188,16 +199,23 @@ We need to unzip it here if we have not already.
...
+                               )
+  unzip(compressed_file, exdir = data_dir)
 }
 ```
 
 #### Expression data
 
 ```{r}
-northcott_expression_file <- file.path("data", "GSE37382", "GSE37382.tsv")
+northcott_expression_file <- file.path(data_dir,
```
Comment (cansavvy) [inline pathway-analysis/qusage_meta_analysis.Rmd:215]: Could declare a `northcott_dir` but maybe this is fine?

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 6]
```diff
@@ -220,8 +238,10 @@ rm(northcott_exprs_df)
 #### Metadata
 
 ```{r}
-northcott_metadata_file <- file.path("data", "GSE37382", 
-                                     "metadata_GSE37382.tsv")
+northcott_metadata_file <- file.path(data_dir,
+                                     "GSE37382", # Replace with name of your unzipped folder
```
Comment (cansavvy) [inline pathway-analysis/qusage_meta_analysis.Rmd:242]: This wording should also be changed if we don't think users would zip their data, which I don't think they would.

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 7]
```diff
@@ -188,16 +199,23 @@ We need to unzip it here if we have not already.
...
+  unzip(compressed_file, exdir = data_dir)
 }
 ```
 
 #### Expression data
 
 ```{r}
-northcott_expression_file <- file.path("data", "GSE37382", "GSE37382.tsv")
+northcott_expression_file <- file.path(data_dir, 
+                                       "GSE37382", # Replace with name of your unzipped folder
```
Comment (cansavvy) [inline pathway-analysis/qusage_meta_analysis.Rmd:216]: See other comment about zipping wordage.

File: pathway-analysis/qusage_meta_analysis.Rmd [hunk 8]
```diff
@@ -253,7 +273,10 @@ northcott_mat <- northcott_mat[, northcott_metadata_df$refinebio_accession_code]
 #### Expression data
 
 ```{r}
-robinson_expression_file <- file.path("data", "GSE37418", "GSE37418.tsv")
+robinson_expression_file <- file.path(data_dir,
```
Comment (cansavvy) [inline pathway-analysis/qusage_meta_analysis.Rmd:276]: Same comment here. If you decide to do a `northcott_dir` should also do `robinson_dir`.

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 9]
```diff
@@ -97,16 +108,23 @@ not already.
...
+  unzip(compressed_file, exdir = data_dir)
 }
 ```
 
 ### Expression data
 
 ```{r}
-expression_file <- file.path("data", "GSE75574", "GSE75574.tsv")
+expression_file <- file.path(data_dir,
+                             "GSE75574", # Replace with name of your unzipped folder
```
Comment (cansavvy) [inline pathway-analysis/qusage_single_dataset.Rmd:125]: I would suggest just searching for `zip` so you make sure you catch all of these.

File: pathway-analysis/qusage_single_dataset.Rmd [hunk 10]
```diff
@@ -312,7 +337,8 @@ We can look at the distribution of genes in this pathway with the
 We tell this function which pathway to plot with the `path.index` argument.
 
 ```{r}
-grep("ECM_RECEPTOR", names(kegg_pathways))
+grep("ECM_RECEPTOR", # Replace with desired pathway -- names of pathways can be seen using the `names(kegg_pathways)` function
```
Comment (cansavvy) [inline pathway-analysis/qusage_single_dataset.Rmd:340]: Bring wording from here to this section. We should add a little more explanation about what this is doing and how `grep` is looking for a pattern.

File: pathway-analysis/results/Project_GSE37418_up_SHH_lfc2_fdr0_05/enriched_geneset_ap_clusters_GSE37418_up_SHH_lfc2_fdr0_05.txt [hunk 11]
```diff
@@ -14,12 +14,12 @@ GO:0045137	GO:0007548	GO:0046661	GO:0008406	GO:0046660	GO:0046546	GO:0048145
 GO:0008589	GO:0009954	GO:0007224
 GO:0006364	GO:0022613	GO:0042254	GO:0016072
 GO:0042063	GO:0010001
-GO:0001654	GO:0043010	GO:0048880	GO:0150063	GO:0007163
+GO:0001654	GO:0043010	GO:0048880	GO:0150063	GO:0007163	GO:0003382
```
Comment (cansavvy) [inline pathway-analysis/results/Project_GSE37418_up_SHH_lfc2_fdr0_05/enriched_geneset_ap_clusters_GSE37418_up_SHH_lfc2_fdr0_05.txt:17]: Do we know why this is different? Also looks like there's some RDS files missing like `Northcott_SHH-Other_QSarray.RDS` so we know why this is? Is it expected?

File: pathway-analysis/results/qusage_t1-t0_kegg_results.tsv [hunk 12]
```diff
@@ -1,187 +1,187 @@
 pathway.name	log.fold.change	p.Value	FDR
-KEGG_CIRCADIAN_RHYTHM_MAMMAL	-0.1732868614855288	6.471848615552389e-5	0.0042846863211493755
```
Comment (cansavvy) [inline pathway-analysis/results/qusage_t1-t0_kegg_results.tsv:2]: Decimals are different looks like? I don't remember if the seed was set?

File: pathway-analysis/ssgsea_example.Rmd [hunk 13]
```diff
@@ -282,7 +306,7 @@ We'll be using `ggplot2` for plotting, so we'll want our ssGSEA values as a
 ```{r}
 comp_coag_df <- data.frame(
   sample_accession = colnames(ssgsea_results),
-  ssgsea_score = ssgsea_results["KEGG_COMPLEMENT_AND_COAGULATION_CASCADES", ]
+  ssgsea_score = ssgsea_results["KEGG_COMPLEMENT_AND_COAGULATION_CASCADES", ] # Replace <KEGG_COMPLEMENT_AND_COAGULATION_CASCADES> with desired pathway
```
Comment (cansavvy) [inline pathway-analysis/ssgsea_example.Rmd:309]: We should be clear that this is distinct from grep in that you'd have to write the name *exactly* as it is in the matrix (data.frame?).

File: pathway-analysis/ora_with_webgestaltr.Rmd (hunk 3 above)
Comment (cbethell) [reply pathway-analysis/ora_with_webgestaltr.Rmd:163]: Ahh both the directory and file name should end in `fdr0_05`, the reason being that it automatically is changed to this anyway when that chunk is ran, even when the file name ending in `fdr0.05` is given,

File: pathway-analysis/results/Project_GSE37418_up_SHH_lfc2_fdr0_05/enriched_geneset_ap_clusters_GSE37418_up_SHH_lfc2_fdr0_05.txt (hunk 11 above)
Comment (cbethell) [reply pathway-analysis/results/Project_GSE37418_up_SHH_lfc2_fdr0_05/enriched_geneset_ap_clusters_GSE37418_up_SHH_lfc2_fdr0_05.txt:17]: I am not sure why it is appearing as it is in the PRs but all of the RDS files are here locally including `Northcott_SHH-Other_QSarray.RDS` and I am able to download the file from this PR. I have yet to get to the bottom of the additional value there however.

File: pathway-analysis/qusage_replicate_vignette.Rmd [hunk 14]
```diff
@@ -225,7 +248,7 @@ timepoint_table <- qsTable(timepoint_qusage_results,
                            # numPathways let's us snag all tested gene sets
                            number = numPathways(timepoint_qusage_results))
 readr::write_tsv(timepoint_table,
-                 file.path("results", "qusage_t1-t0_kegg_results.tsv"))
+                 file.path(results_dir,"qusage_t1-t0_kegg_results.tsv"))
```
Comment (cansavvy) [inline pathway-analysis/qusage_replicate_vignette.Rmd:251]:
```suggestion
                 file.path(results_dir, "qusage_t1-t0_kegg_results.tsv"))
```

File: pathway-analysis/kegg_ortholog_mapping.Rmd [hunk 15]
```diff
@@ -184,7 +184,8 @@ Let's pick the polymerase pathway, where our success should be pretty obvious
 from the mouse gene symbols.
 
 ```{r}
-kegg_mouse_list[[grep("POLYMERASE", names(kegg_mouse_list))]]
+kegg_mouse_list[[grep("POLYMERASE", # Replace with an exact word or phrase that would filter in your desired pathway(s)
```
Comment (cansavvy) [inline pathway-analysis/kegg_ortholog_mapping.Rmd:187]:
```suggestion
kegg_mouse_list[[grep("POLYMERASE", # Replace with a word (case sensitive) or phrase that would filter in your desired pathway(s)
```

**Pull request merged**